custom_components/bond/fan.py

```diff
@@ -24,10 +42,22 @@ def __init__(self, bond, deviceId):
         """Initialize a Bond Fan"""
         self._bond = bond
         self._deviceId = deviceId
+        self._device = self._bond.getDevice(self._deviceId)
+        self._properties = self._bond.getProperties(self._deviceId)
+        self._speed_list = []
 
-        bondProperties = self._bond.getDevice(self._deviceId)
-
-        self._name = bondProperties['name']
+        if BOND_DEVICE_ACTION_SETSPEED in self._device['actions']:
+            if 'max_speed' in self._properties:
+                self._speed_high = int(self._properties['max_speed'])
+                self._speed_low = int(1)
+                self._speed_list.append(SPEED_LOW)
+                if self._speed_high > 2:
+                    self._speed_list.append(SPEED_MEDIUM)
+                    self._speed_medium = (self._speed_high + 1) // 2
+                    self._speed_list.append(SPEED_MEDIUM)
+                self._speed_list.append(SPEED_HIGH)
```

Comment:
if BOND_DEVICE_ACTION_SETSPEED in self._device['actions']:
                self._speed_low = int(1)
                self._speed_list.append(SPEED_LOW)
                self._speed_medium = int(2)
                self._speed_list.append(SPEED_MEDIUM)
                self._speed_high = int(3)
                self._speed_list.append(SPEED_HIGH)